### PR TITLE
[IE PYTHON] dynamic shape api for python

### PR DESCRIPTION
**Pull request by AlexeyLebedev1**
### Details:
 - *Dynamic shape api for IE PYTHON*

### CODE REVIEW

**Review by akuporos — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
why do we need dynamic shape for old python api?

**Comment by akuporos** [comment]
> why do we need dynamic shape for old python api?

We introduced it in the old python api temporarily for unblocking validation. As it'll introduced in the new python api, we'll remove it from the old one.

**Comment by jane-intel** [comment]
@AlexeyLebedev1 is this PR ready for review

File: inference-engine/ie_bridges/python/tests/test_IENetwork.py [hunk 1]
```diff
@@ -158,6 +158,51 @@ def test_reshape():
...
+    assert function.get_parameters()[0].get_partial_shape() == p_shape
+    assert function.get_results()[0].get_output_partial_shape(0) == p_shape
+
+
+@pytest.mark.ngraph_dependent_test
+def test_incorrect_reshape(device):
+    ie = IECore()
+    if device == "CPU":
+        if ie.get_metric(device, "FULL_DEVICE_NAME") == "arm_compute::NEON":
+            pytest.skip("Can't run on ARM plugin due-to ngraph")
```
Comment (eaidova) [inline inference-engine/ie_bridges/python/tests/test_IENetwork.py:197]: I think better to use pytests.skipif instead skipping logic inside test
https://docs.pytest.org/en/6.2.x/reference.html#pytest-mark-skipif

Comment (akuporos) [reply inference-engine/ie_bridges/python/tests/test_IENetwork.py:197]: the problem is there that ARM is also marked as CPU and we need to distinguish it.
BTW @AlexeyLebedev1 prepared another PR #7392 for skipping several tests on ARM. As soon as it is merged, @AlexeyLebedev1 will update this tests too.

File: inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx [hunk 2]
```diff
@@ -53,6 +52,11 @@ cdef c_map_to_dict(map[string, string] c_map):
     return py_dict
 
 
+cdef expand_dims_to_corresponding_layout(shape, layout):
+    single_axes = [1 for i in range(len(layout) - len(shape))]
```
Comment (eaidova) [inline inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx:56]:
```suggestion
    single_axes = [1] * (len(layout) - len(shape))
```

Comment (AlexeyLebedev1) [reply inference-engine/ie_bridges/python/src/openvino/inference_engine/ie_api.pyx]: Done, thanks)

**Comment by AlexeyLebedev1** [comment]
> @AlexeyLebedev1 is this PR ready for review? Pre-commit is red with severe failures.

Now it's ready for review

**Review by jane-intel — APPROVED** [review]
Looks good to me from the customer perspective

**Pull request merged**